.test-infra/junitxml_report.py

```diff
@@ -41,6 +41,9 @@ def print_testsuite(testsuite):
       if child.tag == 'skipped':
         assert status == ''
         status = 'S'
+      elif child.tag == 'failure':
+        assert status == ''
```

Comment:
Consider explaining the assertion:
`assert status == '', "Multiple test signals for a single testcase."` and adding a comment that `testcase` with no `child` tags is considered passing.